solc/CommandLineInterface.cpp

```diff
@@ -322,15 +333,19 @@ void CommandLineInterface::handleIROptimized(std::string const& _contractName)
 	if (!m_options.compiler.outputs.irOptimized)
 		return;
 
+	std::optional<std::string> const& irOptimized = m_compiler->yulIROptimized(_contractName);
+	if (!irOptimized)
+		return;
+
 	if (!m_options.output.dir.empty())
 		createFile(
 			m_compiler->filesystemFriendlyName(_contractName) + "_opt.yul",
-			m_compiler->yulIROptimized(_contractName)
+			irOptimized.value_or(std::string{})
 		);
 	else
 	{
-		sout() << "Optimized IR:" << std::endl;
-		sout() << m_compiler->yulIROptimized(_contractName) << std::endl;
+		sout() << "Optimized IR:\n";
+		sout() << *irOptimized << std::endl;
```

Comment:
Why change half of it to ``\n``, but keep the other one as ``std::endl`` :-)? This can have weird consequences actually...

Reply:
What weird consequences would these by, isn't this just getting rid of one implicit flush? :)

Reply:
I don't think one flush changes much but it's also a pretty random change :)

Reply:
Yeah, true! Not sure what hit me besides flushing too often is a performance drain - but it shouldnt be in this PR if at all. I've reverted the change.